File: sorting/bubble_sort/bubble_sort.cpp
#include <iostream>
#include <vector>

void bubbleSort(std::vector<int>& nums) {
	size_t n = nums.size();
	for (size_t i = 0; i < n - 1; ++i) {
		for (size_t j = 0; j < n-i-1; ++j) {
			if (nums[j] > nums[j+1]){
				std::swap(nums[j], nums[j+1]);
			}
		}
	}
}

File: sorting/bubble_sort/bubble_sort.go
package bubble_sort

func bubbleSort(nums []int) {
	n := len(nums)

	for i := 0; i < n-1; i++ {
		for j := 1; j < n; j++ {
			if nums[i] < nums[j] {
				nums[i], nums[j] = nums[j], nums[i]
			}
		}
	}
}
